provider/kubernetes/client.go

```diff
@@ -155,10 +196,23 @@ func (c *clientImpl) WatchAll(namespaces Namespaces, stopCh <-chan struct{}) (<-
 // GetIngresses returns all Ingresses for observed namespaces in the cluster.
 func (c *clientImpl) GetIngresses() []*extensionsv1beta1.Ingress {
 	var result []*extensionsv1beta1.Ingress
-	for ns, factory := range c.factories {
-		ings, err := factory.Extensions().V1beta1().Ingresses().Lister().List(c.ingressLabelSelector)
+	namespaceList, err := c.GetNamespaces()
+	if err != nil {
+		log.Errorf("could not list namespaces: %v", err)
+		return result
+	}
+
+	var namespaces Namespaces
+	for _, item := range namespaceList.Items {
+		namespaces = append(namespaces, item.ObjectMeta.Name)
+	}
+	for _, ns := range namespaces {
+		log.Debugf("ns: %v, sel: %v", ns, c.ingressLabelSelector)
+		log.Debugf("factories: %+v", c.factories)
```

Comment:
We probably want to drop this debug message (and possibly a few others as well) before merging.